src/string.cr

```diff
@@ -1742,6 +1742,18 @@ class String
     end
   end
 
+  # Returns a string where all chars in the given hash are replaced
+  # by the corresponding *tuple* values.
+  #
+  # ```
+  # "hello".gsub({'e' => 'a', 'l' => 'd'}) # => "haddo"
+  # ```
+  def gsub(tuple : NamedTuple)
+    gsub do |char|
+      tuple[char.to_s]? || char
```

Comment:
Should I change something here or do you consider it a separate change? (I think I do personally). Note that the same pattern is currently in use for the existing `Hash(Char, _)` overload.